### PR TITLE
refine: modify interface for build writer to make it more easy to use

### PR DESCRIPTION
**Pull request by ZENOTME**

### CODE REVIEW

File: icelake/src/io/writer_builder.rs [hunk 1]
```diff
@@ -142,7 +143,7 @@ impl<L: FileAppenderLayer<DefaultFileAppender>> WriterBuilder<L> {
     pub async fn build_equality_delta_writer(
         self,
         unique_column_ids: Vec<usize>,
-    ) -> Result<EqualityDeltaWriter<impl FileAppenderFactory>> {
+    ) -> Result<EqualityDeltaWriter<FileAppenderBuilder<L>>> {
```
Comment (ZENOTME) [inline icelake/src/io/writer_builder.rs:146]: It also may can be replaced as 
`pub async fn build_equality_delta_writer<F: FileAppenderFactory<L::R>>(self,unique_column_ids: Vec<usize>) -> Result<EqualityDeltaWriter<F>`

But the WriterBuilder always use FileAppenderBuilder, so I return FileAppenderBuilder directly.

And I'm concerning do we realy need abtract FileAppenderFactory🤔.

Comment (liurenjie1024) [inline icelake/src/io/writer_builder.rs:146]: Is it possible to make the return type:
```
Box<dyn FileAppdnerFactory<R=Box<dyn FileAppender>>
```

I don't want to leak `FileAppenderBuilder` outside.

**Review by liurenjie1024 — COMMENTED** [review]
Generally LGTM

File: icelake/src/io/writer_builder.rs (hunk 1 above)
Comment (liurenjie1024) [reply icelake/src/io/writer_builder.rs:146]: Yes, maybe we should remove this `FileAppenderFactory` trait given there is not other implementations.

Comment (ZENOTME) [reply icelake/src/io/writer_builder.rs]: After remove `FileAppenderFactory`, we can just expose the `FileAppenderLayer`.

**Review by Xuanwo — APPROVED** [review]

**Review by liurenjie1024 — APPROVED** [review]
LGTM, Thanks!

**Pull request merged**